### PR TITLE
Today App Widget

### PR DESCRIPTION
**Pull request by TinasheMzondiwa**
# What did you change:

WIP: Today App Widget

- [x] Plain Widget
- [x] With Featured Image Widget
- [x] Launch Read screen on Widget tap
- [ ] Logo styling (Plain Widget)
- [ ] ~Rounded corners for Image Widget~
- [ ] Refresh widgets on App launch

# Screenshot/GIFs of this change

###
| Pixel 2
| ----------------------------- |
| Pixel 2

# Merge checklist

- [x] Automated (unit/ui) tests
- [x] Manual tests

### CODE REVIEW

File: common/core/src/main/java/com/cryart/sabbathschool/core/navigation/Destination.kt [hunk 1]
```diff
@@ -36,3 +39,20 @@ enum class Destination(val key: String) {
...
+/**
+ * Builds a navigation [Uri] from a [Destination]
+ *
+ * Example:
+ * Passing [Destination.READ] with extras [Pair("lesson_index", "index")]
+ * will return `ss_app://read?lesson_index=index`
+ *
+ */
+fun Destination.toUri(vararg extras: Pair<String, String> = emptyArray()): Uri {
+    val query = extras.joinToString(separator = "?", prefix = "?") { pair ->
```
Comment (merlin-zaraza) [inline common/core/src/main/java/com/cryart/sabbathschool/core/navigation/Destination.kt]: I think separator should be `&`. Please add test when construct url with more than one parameter.

File: common/lessons-data/src/main/java/app/ss/lessons/data/model/TodayModel.kt [hunk 2]
```diff
@@ -0,0 +1,9 @@
+package app.ss.lessons.data.model
+
+data class TodayModel(
```
Comment (merlin-zaraza) [inline common/lessons-data/src/main/java/app/ss/lessons/data/model/TodayModel.kt]: I'd name it `TodayData` or just `Today`, it's already inside `model` package.

**Comment by VitalikL** [comment]
@TinasheMzondiwa On the screenshot the top widget's rounded corners look stretched

**Comment by TinasheMzondiwa** [comment]
> @TinasheMzondiwa On the screenshot the top widget's rounded corners look stretched 

Rounded corners are still work in progress on the image widget

**Review by merlin-zaraza — APPROVED** [review]

**Comment by VitalikL** [comment]
@TinasheMzondiwa I think we should also provide minimum size for the widget, to avoid this issue

Screenshot_1625939487

**Comment by VitalikL** [comment]
I am also thinking since there are problems with rounding corners for non-image Today Widget we can position the logo inside, like so:

Group 3 Copy

**Pull request merged**